server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -3781,6 +3781,51 @@ protected long doGenerateSeqNoForOperation(Operation operation) {
         }
     }
 
+    public void testSegmentContainsOnlyNoOps() throws Exception {
+        Engine.NoOpResult noOpResult = engine.noOp(new Engine.NoOp(1, primaryTerm.get(), REPLICA, randomNonNegativeLong(), "test"));
+        assertThat(noOpResult.getFailure(), nullValue());
+        engine.refresh("test");
+        Engine.DeleteResult deleteResult = engine.delete(replicaDeleteForDoc("id", 1, 2, randomNonNegativeLong()));
+        assertThat(deleteResult.getFailure(), nullValue());
+        engine.refresh("test");
+    }
+
+    public void testRandomOperations() throws Exception {
```

Comment:
can you add a comment as to why we have this test? something like "a simple test to check that random combination of operations can coexist in segments and be read. This is needed as some fields in lucene may not exist if a segment misses operation types and the code needs to check for that"